src/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tools/Internal/PrecompileDispatchCommand.cs

```diff
@@ -0,0 +1,164 @@
+
+            if (Options.ConfigureCompilationType.HasValue())
+            {
+                dispatchArgs.Add("--configure-compilation-type");
+                dispatchArgs.Add(Options.ConfigureCompilationType.Value());
+            }
+
+            var toolName = typeof(Design.Program).GetTypeInfo().Assembly.GetName().Name;
+            var dispatchCommand = DotnetToolDispatcher.CreateDispatchCommand(
+                dispatchArgs,
+                TargetFramework,
+                Configuration,
+                outputPath: outputPaths.RuntimeOutputPath,
+                buildBasePath: null,
+                projectDirectory: ProjectPath,
+                toolName: toolName);
+
+            var commandExitCode = dispatchCommand
+                .ForwardStdErr(Console.Error)
+                .ForwardStdOut(Console.Out)
+                .Execute()
+                .ExitCode;
+
+            return commandExitCode;
+        }
+
+        private string GetProjectPath()
+        {
+            string projectPath;
+            if (!string.IsNullOrEmpty(Options.ProjectArgument.Value))
+            {
+                projectPath = Path.GetFullPath(Options.ProjectArgument.Value);
+                if (string.Equals(Path.GetFileName(ProjectPath), "project.json", StringComparison.OrdinalIgnoreCase))
+                {
+                    projectPath = Path.GetDirectoryName(ProjectPath);
+                }
+
+                if (!Directory.Exists(projectPath))
+                {
+                    throw new InvalidOperationException($"Could not find directory {projectPath}.");
+                }
+            }
+            else
+            {
+                projectPath = Directory.GetCurrentDirectory();
+            }
+
+            return projectPath;
+        }
+
+        private OutputPaths GetOutputPaths()
+        {
+            var workspace = new BuildWorkspace(ProjectReaderSettings.ReadFromEnvironment());
+
+            var projectContext = workspace.GetProjectContext(ProjectPath, TargetFramework);
+            if (projectContext == null)
+            {
+                Debug.Assert(FrameworkOption.HasValue());
+                throw new InvalidOperationException($"Project '{ProjectPath}' does not support framework: {FrameworkOption.Value()}");
+            }
+
+            var runtimeContext = workspace.GetRuntimeContext(
+                projectContext,
+                RuntimeEnvironmentRidExtensions.GetAllCandidateRuntimeIdentifiers());
+
+            return runtimeContext.GetOutputPaths(Configuration);
+        }
+
+        private NuGetFramework GetTargetFramework()
+        {
+            if (!FrameworkOption.HasValue())
+            {
+                throw new Exception($"Option {FrameworkOption.Template} does not have a value.");
+            }
+
+            return NuGetFramework.Parse(FrameworkOption.Value());
+        }
+
+        private string GetOutputPath()
+        {
+            if (!OutputPathOption.HasValue())
```

Comment:
Would prefer to stay consistent with the Design pattern where you have a single `ParseArguments` method to validate input into the program.